float32 value
float64 time